scripts/filter_dataset.py

```diff
@@ -0,0 +1,179 @@
+    for split in dataset:
+        if "messages" in dataset[split].column_names:
+            dataset[split] = dataset[split].remove_columns("messages")
+    
+    dataset = dataset.map(apply_chat_template, fn_kwargs={"tokenizer": tokenizer})
+    llm = LLM(
+        model=model_args.model_name_or_path,
+        revision=model_args.model_revision,
+        trust_remote_code=model_args.trust_remote_code,
+    )
+
+    sampling_params=SamplingParams(
+        temperature=training_args.temperature,
+        n=training_args.num_generations,
+        max_tokens=training_args.max_completion_length,
+    )
+    
+    def batch_score(examples):
+        prompts = examples["prompt"]
+        
+        outputs = llm.generate(
+            prompts,
+            sampling_params=sampling_params,
+            use_tqdm=False,
+        )
+        repeated_prompts = []
+        reward_completions = []
+        grouped_completions = []
+        for output in outputs:
+            prompt = output.prompt
+            group = []
+            for completion in output.outputs:
+                text = completion.text
+                group.append(text)
+                message = [{"role": "assistant", "content": text}]
+                repeated_prompts.append(prompt)
+                reward_completions.append(message)
+            grouped_completions.append(group)
+        
+        def repeat_each_element_k_times(list_to_repeat: list, k: int) -> list:
+            return [element for item in list_to_repeat for element in [item] * k]
+        
+        rewards_per_func = torch.zeros(len(repeated_prompts), len(reward_funcs))
+        for i, reward_func in enumerate(reward_funcs):
+            keys = [key for key in examples.data.keys() if key not in ["prompt", "completion"]]
+            reward_kwargs = {key: repeat_each_element_k_times(examples[key], training_args.num_generations) for key in keys}
+            output_reward_func = reward_func(prompts=repeated_prompts, completions=reward_completions, **reward_kwargs)
+            # Convert None values to NaN
+            output_reward_func = [reward if reward is not None else torch.nan for reward in output_reward_func]
+
+            rewards_per_func[:, i] = torch.tensor(output_reward_func, dtype=torch.float32)
+            
+        reshaped_rewards = rewards_per_func.view(-1, training_args.num_generations)
+        
+        examples["filter_generations"] = grouped_completions
+        examples["filter_rewards"] = reshaped_rewards.tolist()
+            
+        return examples
+        
+    dataset = dataset.map(batch_score, batched=True, batch_size=256)
+    
+    
+    if script_args.output_dataset_name is not None:
+        unfiltered_dataset_name = script_args.output_dataset_name
+    else:
+        unfiltered_dataset_name = f"{script_args.dataset_name}-generated"
+    filtered_dataset_name = f"{unfiltered_dataset_name}-filtered"
+    
+    dataset.push_to_hub(unfiltered_dataset_name)
```

Comment:
Instead of having a proliferation of datasets, how about we use dataset configs to denote the unfiltered vs filtered subsets, e.g.:

* `unfiltered` = this one
* `filtered_{script_args.low_reward_threshold}-{script_args.high_reward_threshold}`

That way we can do future filterings and have them all in one place.